frappe/core/doctype/prepared_report/prepared_report.py

```diff
@@ -234,6 +235,14 @@ def create_json_gz_file(data, dt, dn, report_name):
 	encoded_content = frappe.safe_encode(frappe.as_json(data, indent=None, separators=(",", ":")))
 	compressed_content = gzip.compress(encoded_content)
 
+	try:
+		create_file_for_prepared_report(json_filename, dt, dn, compressed_content)
+	except OperationalError:
+		frappe.db.connect()
+		create_file_for_prepared_report(json_filename, dt, dn, compressed_content)
```

Comment:
@rohitwaghchaure This will just retry it once, but if the DB connection was killed because of a performance problem, it will not get fixed in the second attempt (mostly)


I don't mind merging this as there's no side effect I can think of. Prepared report generation is _mostly idempotent_. I'd still suggest revisiting why the connection went away. The usual suspect: the thread got killed by MySQL or OOMKiller. Logs should give some insight.